tasmota/tasmota_support/support.ino

```diff
@@ -2562,12 +2568,12 @@ bool NeedLogRefresh(uint32_t req_loglevel, uint32_t index) {
   return ((line - TasmotaGlobal.log_buffer) < LOG_BUFFER_SIZE / 4);
 }
 
-bool GetLog(uint32_t req_loglevel, uint32_t* index_p, char** entry_pp, size_t* len_p) {
-  if (!TasmotaGlobal.log_buffer) { return false; }  // Leave now if there is no buffer available
-  if (TasmotaGlobal.uptime < 3) { return false; }   // Allow time to setup correct log level
+uint32_t GetLog(uint32_t req_loglevel, uint32_t* index_p, char** entry_pp, size_t* len_p) {
```

Comment:
Signature change. What impact has this on caller? Did you verify each?

Reply:
No impact. All callers checked.

Reply:
Changing return type from bool to int should at least produce a warning from the compiler.

This relies on 0 and 1 being mapped to certain Boolean values. That is not clean.

Reply:
There is no warning. Checking a non boolean value for zero or not is very often used and always clean. [More info...](https://en.cppreference.com/w/cpp/language/implicit_conversion#Boolean_conversions)